### PR TITLE
Feature: Enable to change Suite after initialization object

### PR DESCRIPTION
**Pull request by spolischook**
This is needed for change Suite after it's creation.

### CODE REVIEW

File: src/Behat/Testwork/Suite/Suite.php [hunk 1]
```diff
@@ -48,4 +48,12 @@ public function hasSetting($key);
...
      */
     public function getSetting($key);
+
+    /**
+     * Set key value pair to the settings
+     *
+     * @param string $key
+     * @param mixed $value
+     */
+    public function setSetting($key, $value);
```
Comment (Taluu) [inline src/Behat/Testwork/Suite/Suite.php]: That's a BC break. Do not add methods to the interface (the class is OK though)

Comment (spolischook) [reply src/Behat/Testwork/Suite/Suite.php]: @Taluu I know that this is Bc break, but it's more convenient to relay on interface

**Comment by spolischook** [comment]
@Taluu In our case we are redistribute suites after it's configurations according to statistic of suite duration. So we are make auto generated suites that will execute approx same time, see [OroBehatStatisticExtension](https://github.com/oroinc/platform/tree/master/src/Oro/Bundle/TestFrameworkBundle/BehatStatisticExtension)

File: src/Behat/Testwork/Suite/Suite.php (hunk 1 above)
Comment (Taluu) [reply src/Behat/Testwork/Suite/Suite.php]: It may be more convenient but it's still a BC break, so you'll have to comment out this one.

**Comment by everzet** [comment]
@spolischook I know I'm late with comment, but as @Taluu said, we need to respect BC.

**Comment by spolischook** [comment]
BC break removed.

**Comment by ciaranmcnulty** [comment]
@spolischook Can you add a new interface that extends Suite, and have GenericSuite implement it? This would mean you can rely on an interface still

Just adding it to the concrete class risks us breaking your code in future by not respecting the BC

**Comment by pamil** [comment]
@spolischook have you seen if https://github.com/FriendsOfBehat/SuiteSettingsExtension addresses your issue?

**Pull request closed without merging**